File: ZCMS_FrontEnd/src/app/model/Content.ts
import {Field} from "./Field";
import {Status} from "../enum/Status";
import {User} from "./User";
import {ContentModel} from "./ContentModel";

export class Content{
  public id: string;
  public title: string;
  public fields: Field[];
  public createdDate: Date;
  public updatedDate: Date;
  public publishStatus: Status;
  public contentModel: ContentModel;
  public createdUser: User;

  constructor() {
  }
}

File: ZCMS_Backend/src/main/java/com/devz/zcms/entity/Content.java
package com.devz.zcms.entity;

import com.devz.zcms.enumumeration.Status;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;

@Document(collection = "content")
@Data
public class Content {
    @Id
    private String id;
    private String title;
    private List<Field> fields;
    private Date createdDate;
    private Date updatedDate;
    private Status publishStatus;
    @DBRef
    private ContentModel contentModel;
    @DBRef
    private User createdUser;
}
